openvm/extensions/hints-transpiler/src/lib.rs

```diff
@@ -0,0 +1,68 @@
+use openvm_instructions::{
+    instruction::Instruction, riscv::RV32_REGISTER_NUM_LIMBS, LocalOpcode, PhantomDiscriminant,
+};
+use openvm_instructions_derive::LocalOpcode;
+use openvm_stark_backend::p3_field::PrimeField32;
+use openvm_transpiler::{TranspilerExtension, TranspilerOutput};
+use powdr_openvm_hints_guest::{HintsFunct7, HINTS_FUNCT3, OPCODE};
+use rrs_lib::instruction_formats::RType;
+use strum::{EnumCount, EnumIter, FromRepr};
+
+#[derive(
+    Copy, Clone, Debug, PartialEq, Eq, PartialOrd, Ord, EnumCount, EnumIter, FromRepr, LocalOpcode,
+)]
+#[opcode_offset = 0x800]
+#[repr(usize)]
+pub enum HintsOpcode {
+    HINTS,
+}
+
+#[derive(Copy, Clone, Debug, PartialEq, Eq, FromRepr)]
+#[repr(u16)]
+pub enum HintsPhantom {
+    // idk if there is a "proper" way for avoiding conflicts in this number,
+    // just looked at ovm code and picked the next range that didn't seem to be
+    // used
```

Comment:
We do something similar for global opcodes so I'd say it's fine. For opcodes, execution fails if two different executors handle the same opcode, maybe it's the same here?